helix-term/src/ui/statusline.rs

```diff
@@ -66,10 +66,27 @@ pub fn render(context: &mut RenderContext, viewport: Rect, surface: &mut Surface
         append(&mut context.parts.right, text, &base_style, style)
     };
 
+    let statusline = &context.editor.config().statusline;
+    let file_modified = &helix_view::editor::StatusLineElement::FileModificationIndicator;
+
+    let has_modified_element = statusline.left.contains(file_modified)
+        || statusline.center.contains(file_modified)
+        || statusline.right.contains(file_modified);
+
+    let default_file_modified_element = |element_ids: &Vec<StatusLineElementID>| {
+        let mut ret_val = element_ids.to_vec();
+        let file_name_index = element_ids
+            .iter()
+            .position(|id| id == &helix_view::editor::StatusLineElement::FileName);
+        if !has_modified_element && file_name_index != None {
+            ret_val.insert(file_name_index.unwrap() + 1, *file_modified);
+        }
+        return ret_val;
+    };
+
     // Left side of the status line.
```

Comment:
Determining if the statusline is missing the element is not necessary: the new element can be added to the `Default` implementation for `StatusLineConfig` for `left` (in `helix-view/src/editor.rs`) next to `FileName` to emulate the old behavior. It should be possible (although not advisable) to exclude this element in a statusline configuration

Reply:
Ok, I can do that, I was just trying to not break current custom configs.

Reply:
@the-mikedavis I've reverted my changes to the render function on statusline.rs and added the modification indicator to the default config.